Microsoft.Toolkit/Helpers/NotifyTaskCompletion.cs

```diff
@@ -0,0 +1,140 @@
+            }
+            else if (task.IsFaulted)
+            {
+                PropertyChanged.Invoke(this, new PropertyChangedEventArgs(nameof(IsFaulted)));
+                PropertyChanged.Invoke(this, new PropertyChangedEventArgs(nameof(Exception)));
+                PropertyChanged.Invoke(this, new PropertyChangedEventArgs(nameof(InnerException)));
+                PropertyChanged.Invoke(this, new PropertyChangedEventArgs(nameof(ErrorMessage)));
+            }
+            else
+            {
+                PropertyChanged.Invoke(this, new PropertyChangedEventArgs(nameof(IsSuccessfullyCompleted)));
+                PropertyChanged.Invoke(this, new PropertyChangedEventArgs(nameof(Result)));
+            }
+        }
+
+        /// <summary>
+        /// Gets the task that is being waited on.
+        /// </summary>
+        public Task<TResult> Task { get; private set; }
+
+        /// <summary>
+        /// Gets the task wrapper task.
+        /// </summary>
+        public Task TaskCompletion { get; private set; }
+
+        /// <summary>
+        /// Gets the result of the given task.
+        /// </summary>
+        public TResult Result
+        {
+            get
+            {
+                return (Task.Status == TaskStatus.RanToCompletion) ?
+                    Task.Result :
+                    default(TResult);
+            }
+        }
+
+        /// <summary>
+        /// Gets the status of the task.
+        /// </summary>
+        public TaskStatus Status => Task.Status;
+
+        /// <summary>
+        /// Gets a value indicating whether the task is completed.
+        /// </summary>
+        public bool IsCompleted => Task.IsCompleted;
+
+        /// <summary>
+        /// Gets a value indicating whether the task is not completed.
+        /// </summary>
+        public bool IsNotCompleted => !Task.IsCompleted;
+
+        /// <summary>
+        /// Gets a value indicating whether the task was successfully completed.
+        /// </summary>
+        public bool IsSuccessfullyCompleted => Task.Status == TaskStatus.RanToCompletion;
+
+        /// <summary>
+        /// Gets a value indicating whether the task was cancelled.
+        /// </summary>
+        public bool IsCanceled => Task.IsCanceled;
+
+        /// <summary>
+        /// Gets a value indicating whether there was an error with the task.
+        /// </summary>
+        public bool IsFaulted => Task.IsFaulted;
+
+        /// <summary>
+        /// Gets the exception which occured on the task (if one occurred).
+        /// </summary>
+        public AggregateException Exception => Task.Exception;
+
+        /// <summary>
+        /// Gets the inner exception of the task.
+        /// </summary>
+        public Exception InnerException => Exception?.InnerException;
+
+        /// <summary>
+        /// Gets the error message of the task.
+        /// </summary>
+        public string ErrorMessage => InnerException?.Message;
```

Comment:
Realized this on something else I was doing, if no InnerException, should grab message from main exception:

```suggestion
        public string ErrorMessage => InnerException?.Message ?? Exception.Message;
```